src/plugins/discover/public/locator.ts

```diff
@@ -150,6 +156,14 @@ export class DiscoverAppLocatorDefinition implements LocatorDefinition<DiscoverA
     if (viewMode) appState.viewMode = viewMode;
     if (hideAggregatedPreview) appState.hideAggregatedPreview = hideAggregatedPreview;
 
+    if (dataViewSpec) {
+      const [, discoverPlugins] = await this.deps.core.getStartServices();
+      const dataViews = discoverPlugins.data.dataViews;
+
+      const dataView = await dataViews.create(dataViewSpec);
```

Comment:
https://github.com/elastic/kibana/pull/140726/files#diff-b8f2d5620e870f85cae353f721309d67863ee48b4e623d6b585a638ed1b87d90R178
```
state: { dataViewSpec },
```

Reply:
@ppisljar but then it would currently become part of our state/URL and we could used it to create the ad hoc one. didn't we want to omit adding the spec to the URL?  thx!
The link you're posted doesn't seem to work?

Reply:
I think the last commit broke the ad-hoc behavior, when I create a new one, it's no longer selected in the data view picker, what's more I think when navigating back from Lens, the index of the newly created data view has to be replaced in URL to work correctly